aptos-move/framework/aptos-token-objects/sources/token.move

```diff
@@ -77,25 +97,86 @@ module aptos_token_objects::token {
         creator_address: address,
         collection_name: String,
         description: String,
-        name: String,
+        name_prefix: String,
+        name_with_index_suffix: Option<String>,
```

Comment:
Does this have to be Option<>? Why not just allowing passing empty string if there's no suffix? This would eliminate some of the option::none checks below to make the code simpler

Reply:
we use Option<> here to differentiate whether to create a numbered token or not. (this is private internal create_common function - external APIs are clear, and numbered token doesn't have an option for the suffix.